core/src/main/scala/kafka/coordinator/group/DelayedJoin.scala

```diff
@@ -41,3 +44,32 @@ private[group] class DelayedJoin(coordinator: GroupCoordinator,
   override def onExpiration() = coordinator.onExpireJoin()
   override def onComplete() = coordinator.onCompleteJoin(group)
 }
+
+private[group] class InitialDelayedJoin(coordinator: GroupCoordinator,
+                                        purgatory: DelayedOperationPurgatory[DelayedJoin],
+                                        group: GroupMetadata,
+                                        configuredRebalanceDelay: Int,
+                                        delayMs: Int,
+                                        remainingMs: Int) extends DelayedJoin(coordinator, group, delayMs) {
+
+  override def tryComplete(): Boolean = false
+
+  override def onComplete(): Unit = {
+    group synchronized  {
+      if (group.newMemberAdded) {
+        group.newMemberAdded = false
+        val delay = min(configuredRebalanceDelay, remainingMs)
```

Comment:
If we have already exceeded the `rebalance timeout`, even if the flag is still set, we should still `onComplete` right? For example, with `3` and `10`, we should always complete at the end of period `9 - 10` even if there are still new members added during this period right?